sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/AvailabilityStrategy.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.util.List;
+
+/**
+ * The type Availability strategy.
+ */
+public abstract class AvailabilityStrategy {
```

Comment:
I think we should keep this internal for now. I don't see a huge benefit in exposing this, but I am not sure how can we not expose it. @FabianMeiswinkel  / @xinlian12  / @jeet1995 thoughts?

Reply:
I guess the idea here is customer can have their implementation for the availability strategy. but it will need more API to be exposed to work.  so agree to hide it for now.

No good ideas, maybe just expose each different availabilityStrategy as a final class? I am imagining the parameters can be configured on each one will be quite different.

Reply:
I guess the idea was that in the future customers can implement their own AvailabilityStrategy, but i think it will require more API to expose to work.

no really good idea except maybe just directly exposing the ThresholdBasedAvailabilityStrategy as a final class?

Reply:
I like the package-private constructor idea for `AvailabilityStrategy` - still would require `ThresholdBasedAvailabilityStrategy` to be final and its constructor package-private as well. This way inheritance is restricted to the package and won't allow customer added sub-classes.

Do we plan on providing customers the ability to extend `AvailabilityStrategy` though - was thinking of making it a concrete class but with any overridable methods as no-op (like `JsonSerializable`).